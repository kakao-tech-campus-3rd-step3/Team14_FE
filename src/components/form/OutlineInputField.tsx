import type { ComponentPropsWithoutRef } from 'react';

interface OutlineInputFieldProps extends ComponentPropsWithoutRef<'input'> {
  errorMsg?: string;
  rounded?: 'md' | 'lg' | 'xl' | 'none';
}
/**
 *
 * @param errorMsg
 * 에러 메시지를 입력할 수 있는 필드
 * @param rounded
 * 라운드 처리 여부와 크기를 지정 : md | lg | xl | none
 * @param props
 * 기타 input 속성들
 * @returns
 */
const OutlineInputField = ({ errorMsg, rounded = 'none', ...props }: OutlineInputFieldProps) => {
  return (
    <div className="w-full h-full gap-1">
      <input
        {...props}
        className={`w-full h-[3rem] border border-gray-300 rounded-${rounded} py-2 px-3 focus:outline-primary-300`}
      />
      {errorMsg && <p className="text-sm text-red-500">{errorMsg}</p>}
    </div>
  );
};

export default OutlineInputField;
