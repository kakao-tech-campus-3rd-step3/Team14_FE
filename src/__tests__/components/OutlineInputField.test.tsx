import { render, screen, fireEvent } from '@testing-library/react';
import OutlineInputField from '@/components/form/OutlineInputField';

describe('OutlineInputField 컴포넌트', () => {
  describe('기본 스냅샷 테스트', () => {
    test('기본 스냅샷 테스트', () => {
      const { container } = render(<OutlineInputField />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('동작 테스트', () => {
    test('errorMsg가 주어지면 에러 텍스트가 렌더링된다', () => {
      render(<OutlineInputField errorMsg="에러 메시지" />);
      expect(screen.getByText('에러 메시지')).toBeVisible();
    });

    test('onChange 이벤트로 값이 변경된다', () => {
      render(<OutlineInputField aria-label="닉네임" defaultValue="" />);
      const input = screen.getByLabelText('닉네임') as HTMLInputElement;
      fireEvent.input(input, { target: { value: 'suho' } });
      expect(input.value).toBe('suho');
    });
  });
});
