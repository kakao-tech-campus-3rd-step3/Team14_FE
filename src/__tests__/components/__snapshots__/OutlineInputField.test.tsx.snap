// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OutlineInputField 컴포넌트 > 기본 스냅샷 테스트 > 기본 스냅샷 테스트 1`] = `
<div>
  <div
    class="w-full h-full gap-1"
  >
    <input
      class="w-full h-[3rem] border border-gray-300 rounded-none py-2 px-3 focus:outline-primary-300"
    />
  </div>
</div>
`;
